python/perf-kernels/fused_moe/moe-gemm.py

```diff
@@ -0,0 +1,503 @@
+    # (M, top_k, N, K)
+    b_indexed = b[topk_ids]
+    ref_out = torch.einsum("mek,menk->men", a_expanded, b_indexed)
+    if routed_weight:
+        ref_out *= topk_weights.unsqueeze(-1)
+
+    # Validate correctness
+    torch.testing.assert_close(tri_out, ref_out, atol=1e-2, rtol=1e-2)
+
+
+def get_configs():
+    configs = [
+        {"M": 64, "N": 256, "K": 128, "E": 8, "top_k": 2},
+        {"M": 64, "N": 1792, "K": 1024, "E": 8, "top_k": 2},
+        {"M": 64, "N": 7168, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 128, "N": 7168, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 1024, "N": 7168, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 4096, "N": 7168, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 64, "N": 14336, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 128, "N": 14336, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 256, "N": 14336, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 512, "N": 14336, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 1024, "N": 14336, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 2048, "N": 14336, "K": 4096, "E": 8, "top_k": 2},
+        {"M": 4096, "N": 14336, "K": 4096, "E": 8, "top_k": 2},
+    ]
+    return configs
+
+
+def model_benchmark_configs(args):
+    config_file = args.model_configs
+    configs = get_model_configs(config_path=config_file, model_families=["mistral"], model=args.model)
+    fa_configs = []
+    M = args.M if args.M else 4096  # check size
+    # M, K, N, E, top_k
+
+    for model_name, config in configs.items():
+        N1 = config["intermediate_size"]
+        K1 = config["hidden_size"]
+
+        N2 = config["hidden_size"]
+        K2 = config["intermediate_size"] // 2
+
+        E = 8
+        top_k = 2
+        fa_configs.append((model_name, M, N1, K1, E, top_k))
+        fa_configs.append((model_name, M, N2, K2, E, top_k))
+
+    return fa_configs
+
+
+def run_benchmark(custom, args):
+    print_time = args.return_time
+    routed_weight = args.routed_weight
+    dtype = arg_to_torch_dtype[args.dtype]
+    x_names = ['M', 'N', 'K', 'E', 'top_k']
+    if custom:
+        assert args.M and args.N and args.K and args.E and args.top_k, \
+            "Please provide M, N, K, E, top_k for custom runs."
+        x_vals_list = [(args.M, args.N, args.K, args.E, args.top_k)]
+    else:
+        if args.model:
+            x_vals_list = model_benchmark_configs(args)
+            x_names = ['model', 'M', 'N', 'K', 'E', 'top_k']
+        else:
+            configs = get_configs()
+            x_vals_list = [(cfg['M'], cfg['N'], cfg['K'], cfg['E'], cfg['top_k']) for cfg in configs]
+
+    line_names = ['Time (ms)', 'Bandwidth (GB/s)'] if print_time else ['TFLOPS', 'Bandwidth (GB/s)']
+
+    if print_time:
+        # We'll have 2 lines: 'time' and 'bandwidth'
+        line_vals = ['time', 'bandwidth']
+        line_names = ['Time (ms)', 'Bandwidth (GB/s)']
+    else:
+        line_vals = ['tflops', 'bandwidth']
+        line_names = ['TFLOPS', 'Bandwidth (GB/s)']
```

Comment:
I think it should be easier and sometimes more convenient to print both time and tflops.

Reply:
I can change it to print time, flops, mem bandwidth all at the same time by default